totp/totp.go

```diff
@@ -208,3 +210,29 @@ func Generate(opts GenerateOpts) (*otp.Key, error) {
 
 	return otp.NewKeyFromURL(u.String())
 }
+
+// FixGAuthBug appends an extra parameter delimiter "&" to the URI if necessary.
+//
+// This function provides a workaround for a known bug in Google Authenticator.
+// (See issue #94: https://github.com/pquerna/otp/issues/94#issuecomment-2524954588)
+//
+// The bug affects QR codes generated with certain combinations of standards
+// and parameters, causing them to fail when scanned by Google Authenticator.
+//
+// Use this function if you need to generate a QR code using an external tool
+// (outside this library) and require the URI to be compatible with Google Authenticator.
+// It adjusts the URI to ensure proper functionality.
+func FixGAuthBug(uri string) (string, error) {
+	parsedURL, err := url.Parse(uri)
+	if err != nil {
+		return "", err
+	}
+
+	chunks := strings.Split(parsedURL.RawQuery, "&")
+
+	if !strings.HasPrefix(chunks[len(chunks)-1], "secret=") {
```

Comment:
this looks a little strange (out-of-bounds) when the query is empty - no problem, though, because `len(chunks)` will always be >1 (`Split` only returns a nil slice if both s and sep are empty).

Reply:
This is due to make it work if only "secret" parameter is set. In that case **there's no "&" in the URI**.

Such as below:

```url
otpauth://totp/example.com:test_tail@example.com?secret=JBSWY3DPEHPK3PXP
```

Next, `chunks` contains the raw query value, which are the parameters after "?" in the URI.

- Raw query variations ("?" will be omitted):
   - ?&lt;empty&gt;
   - ?issuer=Example
   - ?secret=JBSWY3DPEHPK3PXP
   - ?secret=JBSWY3DPEHPK3PXP&issuer=Example
   - ?issuer=Example&secret=JBSWY3DPEHPK3PXP
Thus, `chunks[len(chunks)-1` ensures to work only and only if the last query parameter is "secret=".

I added more tests for this.

Reply:
For me, the PR was fine. It was just that line that had a strange look to it. If `chunks` was empty, then `chunks[len(chunks)-1]` would become `chunks[-1]`. But it won't because of `Split`'s properties.

Reply:
@yogo1212 

Indeed it won't. But thanks for pointing that out to clarify!